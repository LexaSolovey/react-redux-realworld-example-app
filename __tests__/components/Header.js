import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Header from '../../src/components/Header';

describe('>> ListPagination component', () => {
    
    it('renders LoggedOutView in case when userInfo is not exist', () => {
        const wrapper = renderer.create(
            <MemoryRouter>
<               Header appName={'conduit'} />
            </MemoryRouter>
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders LoggedInView in case when userInfo is exist', () => {
        const currentUser = {
            username: 'xoxach',
            image: 'C:/Program Files (x86)/Google/Chrome/Application/76.0.3809.100/Locales/am.pak'
        };
        const wrapper = renderer.create(
            <MemoryRouter>
                <Header appName={'conduit'} currentUser={currentUser} />
            </MemoryRouter>
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
})
