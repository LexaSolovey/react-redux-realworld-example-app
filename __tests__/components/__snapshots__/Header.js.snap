// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> ListPagination component renders LoggedInView in case when userInfo is exist 1`] = `
<nav
  className="navbar navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      conduit
    </a>
    <ul
      className="nav navbar-nav pull-xs-right"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/editor"
          onClick={[Function]}
        >
          <i
            className="ion-compose"
          />
           New Post
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/settings"
          onClick={[Function]}
        >
          <i
            className="ion-gear-a"
          />
           Settings
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/@xoxach"
          onClick={[Function]}
        >
          <img
            alt="xoxach"
            className="user-pic"
            src="C:/Program Files (x86)/Google/Chrome/Application/76.0.3809.100/Locales/am.pak"
          />
          xoxach
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`>> ListPagination component renders LoggedOutView in case when userInfo is not exist 1`] = `
<nav
  className="navbar navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      conduit
    </a>
    <ul
      className="nav navbar-nav pull-xs-right"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/register"
          onClick={[Function]}
        >
          Sign up
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
